src/sentry/static/sentry/app/stores/alertStore.jsx

```diff
@@ -9,34 +11,61 @@ const AlertStore = Reflux.createStore({
     this.count = 0;
   },
 
-  onAddAlert(message, type, expireAfter, url) {
-    // intentionally recreate array via concat because of Reflux
-    // "bug" where React components are given same reference to tracked
-    // data objects, and don't *see* that values have changed
-    let alertId = this.count++;
+  onAddAlert(alert) {
+    if (defined(alert.id)) {
+      let expirations = getItem('alerts:muted');
+      if (defined(expirations)) {
+        expirations = JSON.parse(expirations);
 
-    this.alerts = this.alerts.concat([{
-      id: alertId,
-      message: message,
-      type: type,
-      url: url
-    }]);
+        // Remove any objects that have passed their mute duration.
```

Comment:
This only runs when alerts are added. In a pathological case the user could dismiss a ton of alerts, and if they never are attempted to be added again, it'll never get cleaned. Maybe this should be cleaned in the store constructor as well?